yang/frr-filter.yang

```diff
@@ -0,0 +1,365 @@
+
+      choice value {
+        description
+          "Standard access list: value to match.
+           Extended access list: source value to match.";
+        mandatory true;
+
+        case host {
+          leaf host {
+            description "Host to match";
+            type inet:ipv4-address;
+          }
+        }
+        case network {
+          leaf network {
+            description "Network to match";
+            type inet:ipv4-prefix;
+          }
+        }
+        case any {
+          leaf any {
+            description "Match any";
+            type empty;
+          }
+        }
+      }
+
+      choice extended-value {
+        when "./sequence >= 100 and ./sequence <= 199 or
+              ./sequence >= 2000 and ./sequence <= 2699";
+        description "Destination value to match";
+
+        case destination-host {
+          leaf destination-host {
+            description "Host to match";
+            type inet:ipv4-address;
+          }
+        }
+        case destination-network {
+          leaf destination-network {
+            description "Network to match";
+            type inet:ipv4-prefix;
+          }
+        }
+        case destination-any {
+          leaf destination-any {
+            description "Match any";
+            type empty;
+          }
+        }
+      }
+    }
+
+    list access-list {
+      description "Access list instance";
+
+      key "type identifier sequence";
+
+      leaf type {
+        description "Access list content type";
+        type enumeration {
+          enum ipv4 {
+            description "Internet Protocol address version 4";
+            value 0;
+          }
+          enum ipv6 {
+            description "Internet Protocol address version 6";
+            value 1;
+          }
+          enum mac {
+            description "Media Access Control address";
+            value 2;
+          }
+
+          /*
+           * Protocol YANG models should augment the parent node to
+           * contain the routing protocol specific value. The protocol
+           * must also augment `value` leaf to include its specific
+           * values or expand the `when` statement on the existing cases.
+           */
+          enum custom {
+            description "Custom data type";
+            value 100;
+          }
+        }
+      }
+
+      leaf identifier {
+        description "Access list identifier";
+        type access-list-name;
+      }
+
+      leaf sequence {
+        description "Access list sequence value";
+        type access-list-sequence;
+      }
+
+      leaf action {
+        description "Access list action on match";
+        type access-list-action;
+        mandatory true;
+      }
+
+      leaf remark {
+        description "Access list remark";
+        type string;
+      }
+
+      choice value {
+        description "Access list value to match";
+        mandatory true;
+
+        case ipv4-prefix {
+          when "./type = 'ipv4'";
```

Comment:
I think this should be `when "../../type = 'ipv4'";`. I'm surprised libyang isn't complaining about this (I'm using the old 0.16-r3 from Debian). Several other `when` statements are using wrong relative paths here and in the route-map module too.

Reply:
`yanglint` complains if I use something else.... Seems `when` behaves differently inside `choice`/`case`.

This is the output I get trying anything else:

```
yanglint -V yang/frr-filter.yang
warn: Schema node "type" not found (../../type) with context node "/frr-filter:filter-list/access-list"
```

or

```
yanglint -V yang/frr-filter.yang                                                        
warn: Schema node "type" not found (../type) with context node "/frr-filter:filter-list/access-list".
```

Reply:
hmm interesting.

I found this small excerpt on the YANG RFC:
>  If the "when" statement is a child of a "uses", "choice", or
>  "case" statement, then the context node is the closest ancestor
>  node to the node with the "when" statement that is also a data
>  node.

So yes, your relative paths are correct :) I'll mark this comment as resolved.